### PR TITLE
Cloudwatch: Use context in aws DescribeLogGroupsWithContext

### PR DESCRIPTION
**Pull request by shabeebk**
**What is this feature?**

In the current way, DescribeLogGroups is used which doesn't allow cancelling the request if the context changes. Using DescribeLogGroupsWithContext is the preferred way.

**Why do we need this feature?**

Using context in the go-based application is always the preferred way wherever it is possible to do so. With this change, we are making the application aware of context when it calls DescribeLogGroupsWithContext.

**Who is this feature for?**

Users

**Which issue(s) does this PR fix?**:

Relates #75498

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/tsdb/cloudwatch/services/log_groups.go [hunk 1]
```diff
@@ -36,7 +38,7 @@ func (s *LogGroupsService) GetLogGroups(req resources.LogGroupsRequest) ([]resou
 	result := []resources.ResourceResponse[resources.LogGroup]{}
 
 	for {
-		response, err := s.logGroupsAPI.DescribeLogGroups(input)
+		response, err := s.logGroupsAPI.DescribeLogGroupsWithContext(context.Background(), input)
```
Comment (iwysiu) [inline pkg/tsdb/cloudwatch/services/log_groups.go:41]: Rather than using a `context.Background()` here, we should get it passed into `GetLogGroups`.

**Review by iwysiu — COMMENTED** [review]
Hi @shabeebk ! Thanks for taking a look at this. I left a comment on `pkg/tsdb/cloudwatch/services/log_groups.go` about passing the context in

File: pkg/tsdb/cloudwatch/services/log_groups.go (hunk 1 above)
Comment (shabeebk) [reply pkg/tsdb/cloudwatch/services/log_groups.go:41]: Thanks @iwysiu for the review. Will update it.

Comment (shabeebk) [reply pkg/tsdb/cloudwatch/services/log_groups.go]: Updated.

**Comment by shabeebk** [comment]
Fixed, rebased, and pushed.

**Review by iwysiu — APPROVED** [review]
Nice work! I'll handle the rest of the CI checks.

**Pull request merged**